src/platform/posix/posix_config.h

```diff
@@ -66,7 +66,7 @@
 
 #ifndef NNG_USE_CLOCKID
 #if defined(__APPLE__)
-#define NNG_USE_CLOCKID CLOCK_REALTIME
+#define NNG_USE_GETTIMEOFDAY
```

Comment:
If this is our "fix" we don't need to alter NNG at all, we just need to adjust our config setup for the build of NNG.
But I would suggest not doing this as it also switches over to using a polling based sleep.

Reply:
Right, but, looking at it, it's not so simple (or I haven't found a simple way) to do it on our side and I was aiming for clarity here.

As for the "polling" based sleep, I'm guessing your concern is that it's different from how it works now (as now it works w/`nano sleep()`)?

Reply:
The problem is that we want to `NNG_USE_GETTIMEOFDAY` only on Mac and that ain't so simple.

I thought of "clock_gettime() of our own", but, wanted to avoid dealing with linkers, toolchains, OS (loader) versions, all which might cause various issues thereof.

Look at this latest change. It's simple and to the point (if a little weird). It will probably never be accepted upstream, but, if we don't care about that at this time, then it's OK.